dask/dataframe/io/parquet.py

```diff
@@ -196,33 +287,43 @@ def _read_pyarrow(fs, paths, file_opener, columns=None, filters=None,
     else:
         out_type = DataFrame
 
-    if index_col and index_col not in all_columns:
-        if isinstance(index_col, list):
-            all_columns = all_columns + index_col
+    if index is False:
+        index_cols = []
+    elif index is None:
+        if has_pandas_metadata:
+            pandas_metadata = json.loads(schema.metadata[b'pandas'].decode('utf8'))
+            index_cols = pandas_metadata.get('index_columns', [])
```

Comment:
As of [this PR](https://github.com/apache/arrow/pull/1271/files) you will need to reconstruct indexes using `pandas_metadata['columns']` to get the actual name of the index (`[c['name'] for c in pandas_metadata['columns']]`) along with `pandas_metadata['index_columns']` which will contain the names of the columns are they are in the `pyarrow.Table` object, which will always match this pattern: `__index_level_\d+__`.

This was done to support the case of an index level name that's the same as a column name.

Reply:
We need a corresponding PR into the pandas developer docs about this

Reply:
Indeed. Incoming.

Reply:
PR here: https://github.com/pandas-dev/pandas/pull/18201

Reply:
The new metadata seems like an improvement in robustness, but from the code it looks like pyarrow can no longer read metadata written using the previous metadata? How do you suggest we handle this transition? We can probably write code that handles both versions, but it's mildly annoying.

Reply:
It would be good to handle both, at least for 1 release cycle. Don't we have integration tests in pandas that can help with this?

Reply:
> It would be good to handle both, at least for 1 release cycle.

Agreed. Was that directed at me or at Phil? From at least reading the arrow code it looks like if I install arrow 0.8.0 (or whatever the next release is) it won't be able to read metadata written with 0.7.1, which makes us supporting both a bit tricky.

Reply:
Hm, I *think* I can do this in a backward compatible way.

Reply:
Let me see if that's possible, and I'll report back.

Reply:
Arrow 0.8.0 isn't released yet, so this can still be sorted out in the next week or so. We can use the pandas integration tests to help validate that all is still good